File: src/main/java/com/akagiyui/drive/component/permission/RequirePermission.java
package com.akagiyui.drive.component.permission;

import com.akagiyui.drive.model.Permission;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * 权限校验注解
 * @author AkagiYui
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface RequirePermission {
    /**
     * 需要的权限
     * @return 权限数组
     */
    Permission[] value() default {};
}

File: src/main/kotlin/com/akagiyui/drive/component/permission/RequirePermission.kt
package com.akagiyui.drive.component.permission

import com.akagiyui.drive.model.Permission

/**
 * 权限校验注解
 *
 * @author AkagiYui
 */
@MustBeDocumented
@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
annotation class RequirePermission(
    vararg val value: Permission = [],
)
